File: coding/Codeforces/round_797/task_A.py
# https://codeforces.com/contest/1690/problem/A

from math import ceil

def solving(n):
    h1 = ceil(n / 3) + 1
    h2 = h1 - 1
    h3 = n - (h1 + h2)
    if h3 == 0:
        h3 = 1
        h2 = h2 -1
    return h2, h1, h3

def main():
    t = int(input())
    answer = []
    for _ in range(t):
        n = int(input())
        h2, h1, h3 = solving(n)
        ans = " ".join(map(str, [h2, h1, h3]))
        answer.append(ans)
    answer = "\n".join(map(str, answer))
    print(answer)

if __name__ == "__main__":
    main()

File: coding/Codeforces/round_925/task_A.cpp
// https://codeforces.com/contest/1931/problem/A
// Created by ruslan515 on 13.02.2024
//

#include<bits/stdc++.h>

using namespace std;

void solving() {
    int n;
    cin >> n;
    int sum, x = -1, y, z;
    for (int i = 1; i <= 26; ++i) {
        for (int j = 1; j <= 26; ++j) {
            for (int k = 1; k <= 26; ++k) {
                sum = i + j + k;
                if (sum == n) {
                    x = i;
                    y = j;
                    z = k;
                    break;
                }
            }
            if (x != -1) {
                break;
            }
        }
        if (x != -1) {
            break;
        }
    }
    vector<int> ans = {x, y, z};
    sort(ans.begin(), ans.end());
    string answer = "";
    for (int el: ans) {
        answer += char (el + 96);
    }
    cout << answer << "\n";

    return;
}

int main() {
    ios::sync_with_stdio(false);
    ios_base::sync_with_stdio(false);
    cin.tie(0);

    int t;
    cin >> t;
    for (int i = 0; i < t; ++i) {
        solving();
    }

    return 0;
}

